src/hooks/progress/use-update-tutorial-progress.ts

```diff
@@ -0,0 +1,186 @@
+import { useEffect } from 'react'
+import { useInView } from 'react-intersection-observer'
+import useAuthentication from 'hooks/use-authentication'
+// Tutorial progress utilities
+import { progressStatusToPercent } from 'lib/learn-client/api/progress/formatting'
+import {
+	useTutorialProgress,
+	useTutorialProgressMutations,
+	TutorialProgressMutationArgs,
+} from 'hooks/progress'
+// Types
+import {
+	TutorialIdCollectionId,
+	TutorialProgressStatus,
+} from 'lib/learn-client/types'
+
+type IntersectionRef = (node?: Element) => void
+
+interface UpdateProgressReturnInterface {
+	startRef: IntersectionRef
+	endRef: IntersectionRef
+}
+
+/**
+ * Given a tutorialId and collectionId,
+ * Return two refs in an object, { startRef, endRef },
+ * which can be placed on elements to track tutorial progress.
+ *
+ * On initial load, regardless of startRef or endRef visibility,
+ * if the tutorial does not yet have recorded progress,
+ * the tutorial will be marked as `visited` (0 percent).
+ *
+ * When the startRef element moves out of the viewport,
+ * the tutorial will be marked as `in_progress` (50 percent).
+ *
+ * When the endRef element moves into the viewport,
+ * the tutorial will be marked as `complete` (100 percent).
+ */
+export function useUpdateTutorialProgress({
+	tutorialId,
+	collectionId,
+}: TutorialIdCollectionId): UpdateProgressReturnInterface {
+	// We shouldn't try to update progress unless we're authenticated
+	const { isAuthenticated } = useAuthentication()
+	// We need to know if progress exists, to know whether to "create" or "update"
+	const { tutorialProgressStatus } = useTutorialProgress({
+		tutorialId,
+		collectionId,
```

Comment:
Should we have this be tied into auth with `enabled`? [See the bookmarks](https://github.com/hashicorp/dev-portal/blob/main/src/hooks/bookmarks/use-is-bookmarked.ts#L32). Or maybe that should be in `useTutorialProgress`.